src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/prototype/AssemblyContigAlignmentSignatureClassifier.java

```diff
@@ -10,174 +9,413 @@
+            return false;
 
-        final AlignmentInterval one = contigWithOnlyOneConfigAnd2Aln.alignmentIntervals.get(0),
-                                two = contigWithOnlyOneConfigAnd2Aln.alignmentIntervals.get(1);
-        return one.referenceSpan.contains(two.referenceSpan)
-                ||
-                two.referenceSpan.contains(one.referenceSpan);
+        return first.forwardStrand ^ last.forwardStrand;
     }
 
-    @VisibleForTesting
-    static boolean hasIncompletePictureDueToOverlappingRefOrderSwitch(final AlignedContig contigWithOnlyOneConfigAnd2AlnToSameChr) {
+    static boolean indicatesIntraChrTandemDupBkpts(final AssemblyContigWithFineTunedAlignments decoratedContig) {
+        return indicatesIntraChrTandemDupBkpts(decoratedContig.contig);
+    }
 
-        final AlignmentInterval one = contigWithOnlyOneConfigAnd2AlnToSameChr.alignmentIntervals.get(0),
-                                two = contigWithOnlyOneConfigAnd2AlnToSameChr.alignmentIntervals.get(1);
-        final SimpleInterval referenceSpanOne = one.referenceSpan,
-                             referenceSpanTwo = two.referenceSpan;
+    static boolean indicatesIntraChrTandemDupBkpts(final AlignedContig contig) {
 
-        if (referenceSpanOne.contains(referenceSpanTwo) || referenceSpanTwo.contains(referenceSpanOne))
-            return true;
+        if (contig.alignmentIntervals.size() != 2)
+            return false;
 
-        // TODO: 10/29/17 this obsoletes the inverted duplication call code we have now, but those could be used to figure out how to annotate which known ref regions are invert duplicated 
-        if (one.forwardStrand != two.forwardStrand) {
-            return referenceSpanOne.overlaps(referenceSpanTwo);
-        } else {
-            if (one.forwardStrand) {
-                return referenceSpanOne.getStart() > referenceSpanTwo.getStart() &&
-                        referenceSpanOne.getStart() <= referenceSpanTwo.getEnd();
-            } else {
-                return referenceSpanTwo.getStart() > referenceSpanOne.getStart() &&
-                        referenceSpanTwo.getStart() <= referenceSpanOne.getEnd();
-            }
+        final AlignmentInterval first = contig.alignmentIntervals.get(0);
+        final AlignmentInterval last = contig.alignmentIntervals.get(1);
+        if ( !first.referenceSpan.getContig().equals(last.referenceSpan.getContig()) )
+            return false;
+
+        if (first.forwardStrand != last.forwardStrand)
+            return false;
+
+        return ChimericAlignment.isLikelySimpleTranslocation(first, last, StrandSwitch.NO_SWITCH);
+    }
+
+    //==================================================================================================================
+    // FOR ASSEMBLY CONTIGS WITH MORE THAN TWO ALIGNMENTS
+
+    /**
+     * convenience struct holding 4 classes:
+     *      1) non-informative contigs who after fine tuning has 0 or 1 good alignment left
```

Comment:
Just for ease of reading it'd be nice to order the lines of this comment in the same order as the fields and constructor params in the code.

Reply:
done with some reordering. The old order was carried-over from the original bad tuple implementation.